sw/include/regs/irq_router.h

```diff
@@ -0,0 +1,30 @@
+// Copyright 2023 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Robert Balas <balasr@iis.ee.ethz.ch>
+// Alessandro Ottaviano <aottaviano@iis.ee.ethz.ch>
+
+// Header definition for irq_router
+// Note that this is not auto-generated due to limitations of reggen
+
+#ifndef _IRQ_ROUTER_REG_DEFS_
+#define _IRQ_ROUTER_REG_DEFS_
+
+// Register width
+#define IRQ_ROUTER_PARAM_REG_WIDTH 32
+
+// Target selection bitmask control register
+#define IRQ_ROUTER_IRQ_TARGET_MASK_REG_OFFSET(i) (0x0 + 0x4 * i)
+
+#define IRQ_ROUTER_NUM_IRQ_SRCS 59
+#define IRQ_ROUTER_NUM_TARGETS 4
+
+enum car_irq_router_target {
+    IRQ_ROUTER_TARGET_NONE          = 0,
```

Comment:
I think the LSB (position 0) is the PLIC (shared among the cores)

Reply:
Out of the 32 possible targets, we have

|---|---|---| --- | ---|---|
| 31 | ... | 3 | 2 | 1 | 0 |
| None | ... | safety island | cva61 | cva60 | plic |

Reply:
Ref

![image](https://github.com/pulp-platform/carfield/assets/62642743/003cc505-cc9d-4b4a-b3af-56d762aaa313)